ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+        }
+        do {
+            generatedGeographicalName = incrementGeneratedGeographicalName(generatedGeographicalName);
+            ucteNodeCode = new UcteNodeCode(
+                    UcteCountryCode.valueOf(country),
+                    substationName + generatedGeographicalName.substring(0, 3),
+                    voltageLevelCodeFromIidmVoltage(voltageLevel.getNominalV()),
+                    generatedGeographicalName.charAt(3)
+            );
+        } while (iidmIdToUcteNodeCodeId.values().contains(ucteNodeCode));
+        iidmIdToUcteNodeCodeId.put(id, ucteNodeCode);
+    }
+
+    /**
+     * create a unique UcteElementId with the given information in parameters
+     *
+     */
+    UcteElementId convertUcteElementId(UcteNodeCode ucteNodeCode1, UcteNodeCode ucteNodeCode2, String id, Terminal terminal1, Terminal terminal2) {
+        if (isUcteNodeId(terminal1.getBusBreakerView().getConnectableBus().getId()) &&
+                isUcteNodeId(terminal2.getBusBreakerView().getConnectableBus().getId())) {
+            return new UcteElementId(
+                    ucteNodeCode1,
+                    ucteNodeCode2,
+                    id.charAt(18));
+        } else {
+            return generateUcteElementId(id, ucteNodeCode1, ucteNodeCode2, null);
+        }
+    }
+
+    /**
+     * increment the last character of the id from 'a' to 'z' until the id is unique, and return it
+     *
+     * @param id the id of the original iidm component to convert
+     * @param ucteNodeCode1 the ucteNodeCode used to create the UcteElementId
+     * @param ucteNodeCode2 the ucteNodeCode used to create the UcteElementId
+     * @return the generated UcteElementId
+     */
+    UcteElementId generateUcteElementId(String id, UcteNodeCode ucteNodeCode1, UcteNodeCode ucteNodeCode2, Character propertyOrderCode) {
+        if (iidmIdToUcteElementId.containsKey(id)) {
+            return iidmIdToUcteElementId.get(id);
+        } else {
+            int orderCode = 'a';
+            UcteElementId ucteElementId = null;
+
+            if (propertyOrderCode != null) { // First we try to generate an ucteElementId with the orderCode in the property
+                ucteElementId = new UcteElementId(ucteNodeCode1, ucteNodeCode2, propertyOrderCode);
+                if (!iidmIdToUcteElementId.values().contains(ucteElementId)) {
+                    iidmIdToUcteElementId.put(id, ucteElementId);
+                    return ucteElementId;
+                }
+            }
+
+            do {
+                if (orderCode <= 'z') {
+                    ucteElementId = new UcteElementId(ucteNodeCode1, ucteNodeCode2, (char) orderCode);
+                } else {
+                    LOGGER.warn("There is more than 26 element with the the id starting by '{}'", ucteElementId);
+                    return null;
+                }
+                orderCode++;
+            } while (iidmIdToUcteElementId.values().contains(ucteElementId));
```

Comment:
Same remark about this do-while loop: it's not a do-while loop because you start with a test!

Reply:
But the test has nothing to do with my loop, the loop condition is `iidmIdToUcteElementId.values().contains(ucteElementId)` the test inside the loop is just to know what I need to do.